packages/client/hooks/useTrebuchetEvents.ts

```diff
@@ -55,16 +56,28 @@ const useTrebuchetEvents = () => {
 
     const onReconnected = () => {
       setConnectedStatus(true)
-      atmosphere.retries.forEach((retry) => retry())
       atmosphere.eventEmitter.emit('removeSnackbar', ({key}) => key === 'offline')
     }
 
     const onData = async (payload: string | object) => {
       // hacky but that way we don't have to double parse huge json payloads. SSE graphql payloads are pre-parsed
       const obj = typeof payload === 'string' ? JSON.parse(payload) : payload
-      if (obj.version && obj.version !== __APP_VERSION__ && 'serviceWorker' in navigator) {
-        const registration = await navigator.serviceWorker.getRegistration()
-        registration?.update().catch()
+
+      // obj.version is sent once on connection, if we have disconnects logged, it's reconnect
+      const isReconnect = obj.version && recentDisconnectsRef.current.length > 0
+
+      if (obj.version && obj.version !== serverVersion.current) {
+        serverVersion.current = obj.version
+        if ('serviceWorker' in navigator) {
+          const registration = await navigator.serviceWorker.getRegistration()
+          registration?.update().catch()
+        }
+
+        if (isReconnect) {
+          atmosphere.retries.clear()
```

Comment:
-1 It didn't occur to me before, but this is wrong. Retries are statically added by hooks
https://github.com/ParabolInc/parabol/blob/a181ff6e9102be689d12641d3bef3a6009097ecd/packages/client/hooks/useQueryLoaderNow.tsx#L27-L30
and just re-executed on each reconnect. That means if we clear the queue here, these will never be re-executed unless the hook is re-evaluated. We should just remove this clear part.

Reply:
@Dschoordsch fixed